operator/src/main/resources/scripts/introspectDomain.py

```diff
@@ -472,17 +458,21 @@ def validateNonDynamicClusterServersHaveSameListenPort(self, cluster):
     firstAdminPortEnabled = None
     for server in self.env.getDomain().getServers():
       if cluster is self.env.getClusterOrNone(server):
-        listenPort = server.getListenPort()
+        listenPort = getRealListenPort(server)
         listenPortEnabled = isListenPortEnabledForServer(server, self.env.getDomain())
         ssl = getSSLOrNone(server)
         sslListenPort = None
         sslListenPortEnabled = None
-        if ssl is not None:
-          sslListenPort = ssl.getListenPort()
-          sslListenPortEnabled = ssl.isEnabled()
-        elif isSecureModeEnabledForDomain(self.env.getDomain()):
-          sslListenPort = 7002
+        ssl_listen_port = getSSLPortIfEnabled(server, self.env.getDomain())
```

Comment:
The logic in getSSLPortIfEnabled is slightly different from the code section that is removed (line 480-484).  Not sure if this difference would cause any problem in some of the use cases.

Reply:
This repeating logic has been refactor to getSSLPortIfEnabled 477 + 480-484.

Reply:
The original code in line 484 set sslListenPort to 7002 if SSL is not None but disabled, while getSSLPortIfEnabled returns None.

Reply:
If SSL is disabled, we shouldn't set it and addSSL only writes out the sslListenPort if it is enabled